### PR TITLE
Refactors inputs and pipe structure to remove redundancy and allow usage of individual vcfs and aggregate vcfs

### PR DESCRIPTION
**Pull request by mcamarad**
## Summary
Refactors pipeline to make it easier to use and more flexible by accepting different types of genomic input data.

## Tasks

- [x] Absorbs https://github.com/lifebit-ai/phewas/pull/23
- [x] Re-structure the pipeline to remove redundancy of processes
- [x] Refactored ind vcf compatibility to work with current pheno and id code files.
- [x] Added support to agg vcf as input
- [x] Updated docs
- [x] Updates tests

## CI tests
https://github.com/lifebit-ai/phewas/actions/runs/638780131

## CloudOS tests
Aggregated vcf: https://cloudos.lifebit.ai/app/jobs/6048a752f164b90113f2d75f
Individual vcf: https://cloudos.lifebit.ai/app/jobs/6048a791f164b90113f2d7f9
Separated plink files: https://cloudos.lifebit.ai/app/jobs/6048adb6f164b90113f2e5a9
Plink files together: https://cloudos.lifebit.ai/app/jobs/6048ad63f164b90113f2e482

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      matrix:
+        nxf_ver: ['20.01.0', '']
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install Nextflow
+        run: |
+          export NXF_VER=${{ matrix.nxf_ver }}
+          wget -qO- get.nextflow.io | bash
+          sudo mv nextflow /usr/local/bin/
+      - name: General test with binary trait
```
Comment (cgpu) [inline .github/workflows/ci.yml:18]:
```Suggestion
Test with a csv file that points to aggregated vcf files splitted by chr  (same as GWAS)
```

File: docs/phewas_pipeline.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+## 1. Requirements
+
+According to https://lifebit.gitbook.io/cloudos/pipelines-documentations-and-examples-1/nextflow-pipelines/phewas and `main.nf` :
+
+### 1.1 - Inputs
+
+- **--vcf_file:** CSV file containing path to VCF files and the following columns:
+
+    1. The `sampleid` column contains a unique sample ID for each individual. These IDs are the same as the IDs used in the file for the `pheno_file` parameter
```
Comment (cgpu) [inline docs/phewas_pipeline.md:13]: Can we parameterise this?
For ukb we can have exported with either `i`, `eid` and for GEL this is platekey column etc, the unique identifier column name can vary.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mcamarad) [reply .github/workflows/ci.yml:18]: Thanks for pointing this out! Already changed! :D

**Review by cgpu — APPROVED** [review]
Approving at this point for merging to dev, as there are a lo of files. Let's catch issues with tests and smaller PRs going forward.

**Pull request merged**